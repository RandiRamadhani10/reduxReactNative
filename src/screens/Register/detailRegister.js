import React from 'react';
import {
  View,
  TouchableOpacity,
  Text,
  StyleSheet,
  Dimensions,
} from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
const screen = Dimensions.get('screen');
const Detailregister = ({navigation}) => {
  return (
    <View style={styles.parent}>
      <Text style={{fontSize: 25, marginBottom: 30}}>
        Registration Complete
      </Text>
      <Icon name="checkmark-circle-sharp" size={100} color="#198754" />
      <Text style={{fontSize: 15, marginTop: 30}}>
        We sent verification to your email
      </Text>
      <TouchableOpacity
        style={{
          padding: 10,
          backgroundColor: 'black',
          borderRadius: 50,
          paddingHorizontal: 30,
          marginTop: 30,
        }}
        onPress={() => {
          navigation.navigate('Login');
        }}>
        <Text style={{color: 'white', fontSize: 15}}>Back to login</Text>
      </TouchableOpacity>
    </View>
  );
};
const styles = StyleSheet.create({
  parent: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
export default Detailregister;
