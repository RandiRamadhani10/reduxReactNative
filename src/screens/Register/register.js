import React, {useState} from 'react';
import {
  View,
  Text,
  TextInput,
  StyleSheet,
  Image,
  Dimensions,
  KeyboardAvoidingView,
  TouchableOpacity,
} from 'react-native';
import {useSelector, useDispatch} from 'react-redux';
import {getLogin} from '../../data/redux/screens/login/loginSlice';
import getRegister from '../../data/redux/screens/register/action';
const screen = Dimensions.get('screen');
const Register = ({navigation}) => {
  const [name, setName] = useState();
  const [email, setEmail] = useState();
  const [password, setPassword] = useState();
  const regis = async () => {
    try {
      const data = {
        email: email,
        password: password,
        name: name,
      };
      const result = await getRegister(data);
    } catch (error) {
      console.log(error);
    } finally {
      navigation.navigate('DetailRegister');
    }
  };
  return (
    <View style={styles.parent}>
      <Image
        style={styles.img}
        source={require('../../assets/logoGedebook.png')}
      />
      <Text style={styles.text}>REGISTER</Text>
      <KeyboardAvoidingView
        style={{width: screen.width * 0.8, alignItems: 'center'}}>
        <TextInput
          testID="fullname-input"
          style={styles.input}
          onChangeText={setName}
          value={name}
          placeholder="FullName"
          keyboardType="default"
        />
        <TextInput
          testID="email-input"
          style={styles.input}
          onChangeText={setEmail}
          value={email}
          placeholder="Email"
          keyboardType="default"
        />
        <TextInput
          testID="password-input"
          style={styles.input}
          onChangeText={setPassword}
          value={password}
          secureTextEntry={true}
          placeholder="Password"
          keyboardType="default"
        />
        <TouchableOpacity
          testID="button-register"
          style={styles.btn}
          onPress={regis}>
          <Text style={{color: 'white', fontSize: 20}}>Register</Text>
        </TouchableOpacity>
      </KeyboardAvoidingView>
      <Text style={{marginTop: 30}}>Already have account ?</Text>
      <TouchableOpacity
        onPress={() => {
          navigation.navigate('Login');
        }}>
        <Text style={{color: 'black', fontSize: 20, fontWeight: 'bold'}}>
          Login
        </Text>
      </TouchableOpacity>
    </View>
  );
};
const styles = StyleSheet.create({
  parent: {flex: 1, justifyContent: 'center', alignItems: 'center'},
  img: {width: screen.width * 0.3, height: screen.width * 0.3},
  text: {fontSize: 20, color: 'black', fontWeight: 'bold'},
  input: {
    width: '100%',
    height: 40,
    margin: 12,
    borderWidth: 1,
    padding: 10,
  },
  btn: {
    padding: 10,
    margin: 5,
    borderRadius: 30,
    backgroundColor: 'black',
    alignItems: 'center',
    width: screen.width * 0.3,
  },
});
export default Register;
