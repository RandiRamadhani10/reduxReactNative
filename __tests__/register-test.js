import 'react-native';
import React from 'react';
import Register from '../src/screens/Register/register';
import {render, fireEvent} from '@testing-library/react-native';

describe('render register screen', () => {
  it('renders correctly', () => {
    const components = render(<Register />).toJSON();
    expect(components).toMatchSnapshot();
  });
});

describe('getButton Register', () => {
  it('button onClick', () => {
    const {getByTestId} = render(<Register />);
    const element = getByTestId('button-register');
    const fullname = 'randi ramadhani';
    const email = 'randi@gmail.com';
    const password = 'pisangisbanana2021';
    const emailButton = getByTestId('email-input');
    const passwordButton = getByTestId('password-input');
    const fullnameButton = getByTestId('fullname-input');

    fireEvent.changeText(fullnameButton, fullname);
    expect(fullnameButton.props.value).toBe(fullname);
    fireEvent.changeText(emailButton, email);
    expect(emailButton.props.value).toBe(email);
    fireEvent.changeText(passwordButton, password);
    expect(passwordButton.props.value).toBe(password);
    fireEvent.press(element)
  });
});
