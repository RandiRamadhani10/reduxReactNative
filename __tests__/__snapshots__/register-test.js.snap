// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render register screen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/logoGedebook.png",
      }
    }
    style={
      Object {
        "height": 225,
        "width": 225,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    REGISTER
  </Text>
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "width": 600,
      }
    }
  >
    <TextInput
      keyboardType="default"
      onChangeText={[Function]}
      placeholder="FullName"
      style={
        Object {
          "borderWidth": 1,
          "height": 40,
          "margin": 12,
          "padding": 10,
          "width": "100%",
        }
      }
      testID="fullname-input"
    />
    <TextInput
      keyboardType="default"
      onChangeText={[Function]}
      placeholder="Email"
      style={
        Object {
          "borderWidth": 1,
          "height": 40,
          "margin": 12,
          "padding": 10,
          "width": "100%",
        }
      }
      testID="email-input"
    />
    <TextInput
      keyboardType="default"
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
      style={
        Object {
          "borderWidth": 1,
          "height": 40,
          "margin": 12,
          "padding": 10,
          "width": "100%",
        }
      }
      testID="password-input"
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "black",
          "borderRadius": 30,
          "margin": 5,
          "opacity": 1,
          "padding": 10,
          "width": 225,
        }
      }
      testID="button-register"
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
          }
        }
      >
        Register
      </Text>
    </View>
  </View>
  <Text
    style={
      Object {
        "marginTop": 30,
      }
    }
  >
    Already have account ?
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Login
    </Text>
  </View>
</View>
`;
